### ISSUE TITLE
MultiSelect disableSelectedItemFiltering doesn't work with searchable prop

### ISSUE THREAD

**Issue by kaureliaExlabs** [mantinedev/mantine#3877]
### What package has an issue

@mantine/core

### Describe the bug

Hi, today I find out that disableSelectedItemFiltering prop doesn't work with searchable prop when you are using groups. Do you have any idea how to fix that?

```tsx
  const testGroupsItems = ["2test", "lorem"];
  const testGroupsItems2 = ["test", "test3", "ipsum"];

<MultiSelect
        label={"Select"}
        placeholder={"Test select"}
        disableSelectedItemFiltering
        searchable
        data={[
          ...testGroupsItems.map((category) => {
            return {
              value: category,
              label: category,
              group: "Group 1"
            };
          }),
          ...testGroupsItems2.map((category) => {
            return {
              value: category,
              label: category,
              group: "Group 2"
            };
          })
        ]}
      />
```

### What version of @mantine/hooks page do you have in package.json?

6.0.4

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

No

**Comment by rtivital** [comment]
Codesandbox with reproduction – https://codesandbox.io/s/naughty-vaughan-z5pqc9?file=/src/App.tsx

**Comment by lgaspari** [comment]
Hi! I've looked into the issue on my end and it seems that when the `disableSelectedItemFiltering` feature was added, it was only implemented for the `not searchable` use case.

I think that the issue can be solved by changing how the second parameter of the `filter()` function ([filter-data:43](https://github.com/mantinedev/mantine/blob/dev/src/mantine-core/src/MultiSelect/filter-data/filter-data.ts#L43)) is calculated from:

```js
filter(
  searchValue,
  value.some((val) => val === data[i].value && !data[i].disabled),
  data[i]
)
```

to:

```js
filter(
  searchValue,
  !disableSelectedItemFiltering &&
    value.some((val) => val === data[i].value && !data[i].disabled),
  data[i]
)
```

That way, if the `disableSelectedItemFiltering` prop is enabled, the `selected` parameter of the `filter()` function will be always `false`; therefore items will never be filtered when selected. This solution should work for groups as well.

I'd create a PR with the fix if that works for you guys, let me know.
